src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqTouchstoneRepository.kt

```diff
@@ -170,6 +171,7 @@ class JooqTouchstoneRepository(
 
         return coverageRecords
                 .filter { it[COVERAGE.ID] != null }
+                .filter{ it[COUNTRY.NAME] != null }
```

Comment:
I think I know what's going on here.  I can't see `COVERAGE.ID` being selected in the new query (which also makes sense since we're now possibly looking at multiple rows), but I think that jooq is interpreting this `ID` field then as any other field with the same name (in this case coverage_set.id) hence why this filter clause stopped working and you had to add the second filter step. 

So you can just delete the first filter step (similarly on line 176)